src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -174,13 +174,18 @@ public JToken TransformContent(ErrorBuilder errors, JsonSchema schema, FilePath
         {
             var href = GetXrefHref(file, uid, uidCount, obj.Parent == null);
             var monikers = _monikerProvider.GetFileLevelMonikers(errors, file);
-            var xref = new InternalXrefSpec(
-                uid,
-                href,
-                file,
-                monikers,
-                obj.Parent?.Path,
-                schema.SchemaType ?? (schema == rootSchema ? _documentProvider.GetMime(file).Value : null));
+
+            string? schemaType = uidSchema.SchemaType;
+            if (schemaType is null && schema.SchemaTypeProperty != null && obj.TryGetValue(schema.SchemaTypeProperty, out var type))
```

Comment:
When `SchemaTypeProperty` exists, should not fallback to mime type in any case, so the structure might look like below. Also consider extracting get schema type into a new method.


```
if (schemaType is null)
{
    if (schemaTypeProperty != null)
    {
    }
    else if (schema == rootSchema)
    {
    }
}
```

Reply:
OK